lib/bundler/definition.rb

```diff
@@ -137,8 +137,17 @@ def resolve_remotely!
     # @return [Bundler::SpecSet]
     def specs
       @specs ||= begin
-        specs = resolve.materialize(Bundler.settings[:cache_all_platforms] ? dependencies : requested_dependencies)
-
+        begin
+          specs = resolve.materialize(Bundler.settings[:cache_all_platforms] ? dependencies : requested_dependencies)
+        rescue GemNotFound => e # Handle yanked gem
+          gem_info = extract_gem_info(e)
+          locked_gem = @locked_specs[gem_info[0]].last
```

Comment:
You mean assign like this?
gem_name = gem_info[0]
gem_version = gem_info[1]
Becasue what I did in extract_gem_info is a regex match